### ISSUE TITLE
Select a subfolder from tab popup with [Enter]

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/QuickFolders#560]
When we use the keyboard to open the dropdown menu on a QuickFolders tab, we want to be able to navigate to a folder by selecting any subfolder menu and pressing [Enter] in order to open the folder. This is especially difficult with menu items that contain subfolders, as in a standard menu these are designed to open the subfolders. In QF I also programmed in the ability to open the folder (and drag mail to) by clicking the menu parent (as it represents a folder):

![Image](https://github.com/user-attachments/assets/40a3020f-d36f-428a-8092-4cb20cb9c7e5)

this was in order to remove the annoying duplication of items in standard folder tree browsers of Thunderbird:

![Image](https://github.com/user-attachments/assets/3276d592-af9c-4429-80a8-a71da487446a)

**Comment by RealRaven2000** [comment]
This is related to #552 (keyboard navigation within QuickFolders toolbar) but it definitely deserves its own issue / topic.

**Comment by RealRaven2000** [comment]
OK, so we need to decide what to do in this case:

![Image](https://github.com/user-attachments/assets/eb50d6f5-e0d3-44fd-867c-980c88d2940c)

When we press Enter it opens the subfolder menu FiltaQuilla and shows the child folders, also focuses the input on the first child folder:

![Image](https://github.com/user-attachments/assets/52ebc178-1a29-4a25-b4b0-f8917e746fdd)

since I do not want to change this (expected / default) behavior I suggest 2 things

- we support using an accelerator combo (let's say Shift+Enter / Ctrl+Enter) to force opening the folder instead of the submenu
- if the menu gets opened with a single Enter, we could add the folder itself to the top for ease of use: (so I guess we return to the behavior of Thunderbird). maybe this could be optional to avoid clutter.

**Comment by RealRaven2000** [comment]
I tried adding a keydown listener for this feature, but it doesn't get triggered at all:


```js
addSubMenuEventListener: function (subMenu, url) {
  // url is specific to this function context so it should be snapshotted from here
  subMenu.addEventListener("click",
    function(evt) {
      QuickFolders.Interface.onSelectParentFolder(url, evt);
    }, false);
// [issue 560]
subMenu.addEventListener(
    "keydown",
      function (evt) {
        if (evt.key == "Enter" && (evt.ctrlKey || evt.shiftKey)) {
          evt.stopPropagation();
          QuickFolders.Interface.onSelectParentFolder(url, evt);
        }
      },
      {capture: true}
    );
} ,
```

so I assume the menu controls handle keyboard input on a much lower level and do not allow the event to propagate!

**Comment by RealRaven2000** [comment]
Added the option for the extra menu item:

![Image](https://github.com/user-attachments/assets/662f7aa5-d7f4-4eef-9e5a-2a0bde02c4a3)

Resulting popup, to open the parent folder with [Enter]:

![Image](https://github.com/user-attachments/assets/867b0a7d-5f53-4a26-a28d-fcb3bef0db62)

test version:

[QuickFolders-mx-6.10.1pre68.zip](https://github.com/user-attachments/files/19374827/QuickFolders-mx-6.10.1pre68.zip)

----

To try out this version, download the zip file and then drag it into Thunderbird Add-ons Manager (without extracting)
for users of screen reader you can also install this via the tools button after the "Find more addons" search text. activate the button with Enter then chose the 3rd menu entry "install addon from file...". Then select the zip file to install.

**Comment by RealRaven2000** [comment]
Implemented in 6.10.1. - Published 20/03/2025
